File: test_job_listings.js
var assert = require('assert'),
test = require('selenium-webdriver/testing'),
webdriver = require('selenium-webdriver');
var driver;

test.before(function() {
  this.timeout(15000);
  driver = new webdriver.Builder().
  withCapabilities(webdriver.Capabilities.chrome()).
  build();
  driver.get('http://www.dokkio.com');
});

test.after(function() {
  driver.quit();
});

test.describe('Dokkio Jobs Section', function() {
  test.it('QA job listing is not displayed until you click QA button', function() {
    var jobHeader1 = driver.findElement(webdriver.By.css('div.iw-so-tabs-panel.iw-so-tab-active h3'));
    var buttonQA = driver.findElement(webdriver.By.css('li#Jobs-2 a'));
    jobHeader1.getText().then(function(value) {
      assert.equal(value, 'Senior Full Stack Engineer');
    });
    buttonQA.click()
    var jobHeader2 = driver.findElement(webdriver.By.css('div.iw-so-tabs-panel.iw-so-tab-active h3'));
    jobHeader2.getText().then(function(value) {
      assert.equal(value, 'QA Engineer');
    });
  });
});

File: test_job_listings.py
import unittest
from selenium import webdriver
from webdriver_manager.chrome import ChromeDriverManager


class DokkioTests(unittest.TestCase):

    @classmethod
    def setUpClass(cls):
        cls.driver = webdriver.Chrome(ChromeDriverManager().install())
        cls.driver.implicitly_wait(30)
        cls.driver.maximize_window()
        cls.driver.get("http://www.dokkio.com")

    def test_qa_job_listing(self):
        self.job_header1 = self.driver.find_element_by_css_selector("div.iw-so-tabs-panel.iw-so-tab-active h3")
        self.button_qa = self.driver.find_element_by_css_selector("li#Jobs-2 a")
        self.assertEqual("Senior Full Stack Engineer", self.job_header1.text)
        self.button_qa.click()
        self.job_header2 = self.driver.find_element_by_css_selector("div.iw-so-tabs-panel.iw-so-tab-active h3")
        self.assertEqual("QA Engineer", self.job_header2.text)

    @classmethod
    def tearDownClass(cls):
        cls.driver.quit()

if __name__ == '__main__':
    unittest.main(verbosity=2)
